### PR TITLE
[14.0][MIG] fetchmail_incoming_log: Migration to 14.0

### PR DESCRIPTION
**Pull request by hkapatel-initos**
Migration to 14.0
prefer EmailMessage over deprecated Message.

### CODE REVIEW

**Comment by hkapatel-initos** [comment]
@MiquelRForgeFlow and @AdriaGForgeFlow can you please help me to review my changes and why Tavis is not green because I have tried multiple times and still test case has been failed and when I test locally No errors appear.
I have changed in message process as par odoo standard.

**Comment by MiquelRForgeFlow** [comment]
@hkapatel-initos error is in tests of test_mail of OCB, do not worry, your PR is okay.

**Review by MiquelRForgeFlow — APPROVED** [review]

**Comment by hkapatel-initos** [comment]
@pedrobaeza  can you please review this pr?

**Comment by MiquelRForgeFlow** [comment]
but not merge yet, let me fix the ci first

**Comment by hkapatel-initos** [comment]
@atchuthan can u please review this pr?

**Comment by hkapatel-initos** [comment]
@AaronHForgeFlow  can you please review this PR?

**Comment by MiquelRForgeFlow** [comment]
@hkapatel-initos please, cherry-pick https://github.com/OCA/server-tools/commit/d0155a74abaf32be8c1c700aa36d4338d3c15a29

**Comment by hkapatel-initos** [comment]
@victoralmau can you please review this pr?

**Comment by MiquelRForgeFlow** [comment]
@hkapatel-initos if you add commit https://github.com/OCA/server-tools/commit/d0155a74abaf32be8c1c700aa36d4338d3c15a29, then ci will be green

File: .travis.yml [hunk 1]
```diff
@@ -14,20 +14,29 @@ addons:
...
-        - TESTS=1 ODOO_REPO="OCA/OCB"
+        - TESTS="1" ODOO_REPO="OCA/OCB"
+          EXCLUDE="fetchmail_incoming_log,fetchmail_notify_error_to_sender_test"
+
 env:
   global:
-    - VERSION="14.0" TESTS="0" LINT_CHECK="0" MAKEPOT="0"
-    - MQT_DEP=PIP
+    - VERSION="13.0" TESTS="0" LINT_CHECK="0" MAKEPOT="0"
+    - COUNTRY="US"
```
Comment (MiquelRForgeFlow) [inline .travis.yml]: this is wrong

Comment (hkapatel-initos) [reply .travis.yml]: @MiquelRForgeFlow Thanks for the review suggestions are implemented you can re-review it

**Review by MiquelRForgeFlow — APPROVED** [review]

**Review by fshah-initos — APPROVED** [review]
LGTM

**Comment by hkapatel-initos** [comment]
@victoralmau can you please review this PR?

File: fetchmail_incoming_log/models/mail_thread.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        message = email.message_from_bytes(message, policy=email.policy.SMTP)
+        msg_dict = self.message_parse(message, save_original=save_original)
+        _logger.info(
+            "Fetched mail from %s to %s with Message-Id %s",
+            msg_dict.get("from"),
+            msg_dict.get("to"),
+            msg_dict.get("message_id"),
+        )
+
+        return super(MailThread, self).message_process(
```
Comment (victoralmau) [inline fetchmail_incoming_log/models/mail_thread.py]:
```suggestion
        return super().message_process(
```

File: .travis.yml [hunk 3]
```diff
@@ -14,16 +14,25 @@ addons:
...
+        - TESTS="1" ODOO_REPO="odoo/odoo" MAKEPOT="1"
     - stage: test
       env:
-        - TESTS=1 ODOO_REPO="OCA/OCB"
+        - TESTS="1" ODOO_REPO="OCA/OCB"
+          EXCLUDE="fetchmail_incoming_log,fetchmail_notify_error_to_sender_test"
+
 env:
   global:
     - VERSION="14.0" TESTS="0" LINT_CHECK="0" MAKEPOT="0"
```
Comment (victoralmau) [inline .travis.yml:38]: IMO isn't a good idea have these changes to prevent errors in OCA/OCB, it's necessary to make all changes to pass all tests.

**Review by victoralmau — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 3 above)
Comment (hkapatel-initos) [reply .travis.yml:38]: yes, it's a good idea so. I add the cherry pic of your commit

Comment (victoralmau) [reply .travis.yml:38]: ok thanks, I have already remembered the reason for this change.

**Review by victoralmau — APPROVED** [review]

**Comment by hkapatel-initos** [comment]
@dreispt It would be great if you could try to merge it .

**Pull request merged**